internal/project/service.go

```diff
@@ -177,13 +179,30 @@ func (s *Service) OpenFile(fileName string, fileContent string, scriptKind core.
 	s.printProjects()
 }
 
-func (s *Service) ChangeFile(fileName string, changes []ls.TextChange) {
+func (s *Service) ChangeFile(document lsproto.VersionedTextDocumentIdentifier, changes []lsproto.TextDocumentContentChangeEvent) error {
+	fileName := ls.DocumentURIToFileName(document.Uri)
 	path := s.toPath(fileName)
-	info := s.GetScriptInfoByPath(path)
-	if info == nil {
-		panic("scriptInfo not found")
+	scriptInfo := s.GetScriptInfoByPath(path)
+	if scriptInfo == nil {
+		return fmt.Errorf("file %s not found", fileName)
+	}
+
+	textChanges := make([]ls.TextChange, len(changes))
+	for i, change := range changes {
+		if partialChange := change.TextDocumentContentChangePartial; partialChange != nil {
```

Comment:
There is delayUpdateProjectGraph in this file- in strada we use to schedule the update (unless new change came in - then we would cancel and restart the timer) - with this change does this mean - send the refresh diagnostics notification - instead of updating as we want to update on next request instead ?

Reply:
We don't send the diagnostic refresh notification from here. The client will ask for new diagnostics after an update, so I don't think there's any need for us to schedule anything.